File: program/program/check-valid-date/check_valid_date.go
package main

import (
	"fmt"
	"time"
)

func main() {
	var input string
	fmt.Printf("Enter date in YYYY-MM-DD format: ")
	fmt.Scan(&input)

	_, err := time.Parse("2006-01-02", input)
	if err != nil {
		fmt.Println("Invalid Date")
	} else {
		fmt.Println("Valid Date")
	}

}

File: program/program/check-valid-date/CheckValidDate.rb
# frozen_string_literal: true

require 'date'

def check_valid_date(date)
  Date.parse(date)
  true
rescue ArgumentError
  false
end

if $PROGRAM_NAME == __FILE__
  print 'Input  : '
  date = gets.chomp
  # Match the regular expression for a date
  if date =~ /\d{4}-\d{2}-\d{2}/
    puts "Output : #{check_valid_date(date)}"
  else
    puts 'Output : Invalid date. Date must be in the format YYYY-MM-DD.'
  end
end
